scripts/ci-entrypoint.sh

```diff
@@ -181,6 +181,9 @@ timeout --foreground 1800 bash -c wait_for_nodes
 
 # install cloud-provider-azure components, if using out-of-tree
 if [[ -n "${TEST_CCM:-}" ]]; then
+    if [[ -n "${TEST_WINDOWS:-}" ]]; then
+        "${KUBECTL}" wait --for=condition=Ready pod -l app=calico -n kube-system --timeout=10m
```

Comment:
Can you incorporate something similar for the capz E2E tests?

```
$ git diff
diff --git a/test/e2e/common.go b/test/e2e/common.go
index 6037616b..00a97ef5 100644
--- a/test/e2e/common.go
+++ b/test/e2e/common.go
@@ -243,6 +243,11 @@ func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCl
        Eventually(func() error {
                return getter.Get(ctx, key, kubeadmControlPlane)
        }, input.WaitForControlPlaneIntervals...).Should(Succeed(), "Failed to get KubeadmControlPlane object %s/%s", cluster.Spec.ControlPlaneRef.Namespace, cluster.Spec.ControlPlaneRef.Name)
+       clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+       workloadClusterClient := clusterProxy.GetClient()
+       for _, ds := range []string{"calico-node-windows", "calico-node"} {
+               WaitForDaemonset(ctx, input, workloadClusterClient, ds, "kube-system")
+       }
        if kubeadmControlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.ControllerManager.ExtraArgs["cloud-provider"] == "external" {
                InstallCloudProviderAzureHelmChart(ctx, input)
        }
```

The above waits for _both_ the Windows *and* Linux daemonset pods, but given that we've already identified edge cases if we install cloud-node-manager-windows before calico-node-windows is ready, there may be similar edge cases to discover if we install cloud-node-manager before calico-node is ready.

@CecileRobertMichon FYI, these changes will need to be reflected in the helm + calico effort if they aren't already

Reply:
> Can you incorporate something similar for the capz E2E tests?

I'm already doing this in #2334 so let's just keep this PR focused on ci-entrypoint.sh